tensorflow/compiler/jit/mark_for_compilation_pass.cc

```diff
@@ -1076,6 +1076,86 @@ StatusOr<bool> IsIdentityDrivingConstsInLoop(Node* node) {
   return true;
 }
 
+const absl::flat_hash_map<string, std::vector<string>> whitelist_table = {
```

Comment:
Please document the format of this table.

We also don't allow non-trivial global destructors because they don't play well with multi-threading.  I think a better phrasing is:

```
absl::flat_hash_map<string, std::vector<string>> *CreateWhitelist() {
  absl::flat_hash_map<string, std::vector<string>>* result = new ...;
  // Use explicit code to populate "result", possibly with comments.
}

const absl::flat_hash_map<string, std::vector<string>>& GetOrCreateWhitelist() {
  static absl::flat_hash_map<string, std::vector<string>>* whitelist = CreateWhitelist();
  return *whitelist;
}
```

Reply:
I fixed the non-trivial global destructors.
I also added documentation of the format. If you wanted more then this, just tell me.

Reply:
I think it's better to have `static` in `CreateWhitelist` in order not accidentally create a leak if someone else calls it, also it seems this is how it is usually done in XLA, e.g. https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/transfer_manager.cc#L41